### ISSUE TITLE
12.0.2 Maximum call stack size exceeded

### ISSUE THREAD

**Issue by astone2014** [stevermeister/ngx-cookie-service#145]
## Expected Behavior

```
npm i ngx-cookie-service@12.0.0 --save
npm i; ng build devicewise-angular --prod
```
successfully build.

## Actual Behavior

```
npm i ngx-cookie-service@latest --save
✖ Compiling TypeScript sources through NGC
ERROR: Maximum call stack size exceeded
An unhandled exception occurred: Maximum call stack size exceeded
```
Maximum call stack size exceeded when running production build with `12.0.2`.

## Steps to Reproduce the Problem

Install `ngx-cookie-service@12.0.2` and run `ng build --prod`

## Specifications

- Version: 12.0.2
- Browser: N/A

**Comment by pavankjadda** [comment]
@astone2014 It works for me as expected. I don't think this has anything to do with this library. 

```
HOSTNAME:my-app jaddap2$ npm i ngx-cookie-service

added 1 package, and audited 1342 packages in 2s

87 packages are looking for funding
  run `npm fund` for details

5 moderate severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
HOSTNAME:my-app jaddap2$ ng build
⠋ Generating browser application bundles (phase: setup)...Compiling @angular/core : es2015 as esm2015
Compiling @angular/common : es2015 as esm2015
Compiling @angular/platform-browser : es2015 as esm2015
Compiling @angular/router : es2015 as esm2015
Compiling @angular/platform-browser-dynamic : es2015 as esm2015
✔ Browser application bundle generation complete.
✔ Copying assets complete.
✔ Index html generation complete.

Initial Chunk Files               | Names         |      Size
main.df1ded17827a01336d91.js      | main          | 214.56 kB
polyfills.31472d1a31387efcb829.js | polyfills     |  36.19 kB
runtime.47c71d9e559014d0f763.js   | runtime       |   1.02 kB
styles.31d6cfe0d16ae931b73c.css   | styles        |   0 bytes

                                  | Initial Total | 251.77 kB

Build at: 2021-08-06T14:40:18.555Z - Hash: c28bbd7e8251a06cada6 - Time: 23198ms

```

Please share reproducible example in StackBlitz

**Comment by astone2014** [comment]
Well I can't reproduce in a clean environment.
Will reopen if I get more info.

**Comment by eydrian** [comment]
I have the same issue. 
It happens when building an angular library where ngx-cookie-service is a dependency and injected in a module as provider, Ivy disabled and the following angularCompilerOptions are set:
```
  "angularCompilerOptions": {
    "compilationMode": "partial",
    "enableIvy": false,
    "skipTemplateCodegen": true,
    "strictMetadataEmit": true,
    "fullTemplateTypeCheck": true,
    "strictInjectionParameters": true,
    "enableResourceInlining": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
  ```

**Comment by astone2014** [comment]
@eydrian Yes, this happened when compiling an angular library that uses ngx-cookie-service as a dependency. Only with production build (ivy disabled).

**Comment by pavankjadda** [comment]
@eydrian @astone2014 I created [sample library](https://github.com/pavankjadda/ngx-cookie-service-bug-145) and imported Ngx Cooki Service. I am able to build it without any issues. See the repo [here](https://github.com/pavankjadda/ngx-cookie-service-bug-145). Check it and let me know if this something you guys are doing.

**Comment by eydrian** [comment]
@pavankjadda I created a PR for the example that reproduces the issue: https://github.com/pavankjadda/ngx-cookie-service-bug-145/pull/1

**Comment by pavankjadda** [comment]
@eydrian @astone2014 version 12.0.3 should fix the issue. Please try and let me know

**Comment by astone2014** [comment]
@pavankjadda  the problem was resolved in 12.0.3.
Thank you! 🎉

**Comment by eydrian** [comment]
I can also confirm from my side, that this issue is resolved. Thanks for taking care of it that quickly!
